ledger/participant-integration-api/src/main/scala/platform/store/dao/events/TransactionIndexing.scala

```diff
@@ -188,15 +149,19 @@ object TransactionIndexing {
 
   final case class EventsInfo(
       events: Vector[(NodeId, Node)],
-      archives: Set[ContractId],
       stakeholders: WitnessRelation[NodeId],
       disclosure: WitnessRelation[NodeId],
   )
 
+  // most involved change: divulgence is now an event instead encoded in contract/contract_witness tables
+  // also we generate these events directly as part of the other events, so we need to land the necessery
+  // data in the EventsTablePostgresql
+  case class TransactionDivulgedContract(contractId: ContractId,
+                                         contractInst: Option[ContractInst], // this we do not necessarily have: for public KV ledgers the divulged contract list is empty, since all contracts must be at all ledger as create nodes already
+                                         visibility: Set[Party])
+
   final case class ContractsInfo(
-      netCreates: Set[Create],
-      netArchives: Set[ContractId],
-      divulgedContracts: Iterable[DivulgedContract],
+      divulgedContracts: Iterable[TransactionDivulgedContract],
   )
 
   final case class ContractWitnessesInfo(
```

Comment:
This name reads wrong to me. It contains no witness information, and stores data about both create and exercise events.

Reply:
It is dead code, removed